src/internal/kopia/base_finder_test.go

```diff
@@ -701,9 +873,167 @@ func (suite *BaseFinderUnitSuite) TestGetBases() {
+					testID3,
+					testT3,
+					testCompleteMan,
+					testBackup3,
+					nil,
+					testMail,
+					testUser1,
+				),
+				newManifestInfo(
+					testID2,
+					testT2,
+					testCompleteMan,
+					testBackup2,
+					nil,
+					testMail,
+					testUser1,
+				),
+				newManifestInfo(
+					testID1,
+					testT1,
+					testCompleteMan,
+					testBackup1,
+					nil,
+					testMail,
+					testUser1,
+				),
+			},
+			expectedBaseReasons: map[int][]Reasoner{
+				2: testUser1Mail,
+			},
+			expectedAssistManifestReasons: map[int][]Reasoner{
+				0: testUser1Mail,
+				2: testUser1Mail,
+			},
+			expectedAssistReasons: map[int][]Reasoner{
+				0: testUser1Mail,
+			},
+			backupData: []backupInfo{
+				newBackupModel(
+					testBackup4,
+					true,
+					true,
+					false,
+					map[string]string{model.BackupTypeTag: model.AssistBackup},
+					nil),
+				newBackupModel(
+					testBackup3,
+					true,
+					true,
+					false,
+					map[string]string{model.BackupTypeTag: model.AssistBackup},
+					nil),
+				newBackupModel(testBackup2, true, true, false, nil, nil),
+				newBackupModel(testBackup1, true, true, false, nil, nil),
+			},
+		},
+		{
+			name:  "Newer merge base than assist base",
+			input: testUser1Mail,
+			manifestData: []manifestInfo{
+				newManifestInfo(
+					testID2,
+					testT2,
+					testCompleteMan,
+					testBackup2,
+					nil,
+					testMail,
+					testUser1,
+				),
+				newManifestInfo(
+					testID1,
+					testT1,
+					testCompleteMan,
+					testBackup1,
+					nil,
+					testMail,
+					testUser1,
+				),
+			},
+			expectedBaseReasons: map[int][]Reasoner{
+				0: testUser1Mail,
+			},
+			expectedAssistManifestReasons: map[int][]Reasoner{
+				0: testUser1Mail,
+			},
+			expectedAssistReasons: map[int][]Reasoner{},
+			backupData: []backupInfo{
+				newBackupModel(testBackup2, true, true, false, nil, nil),
+				newBackupModel(
+					testBackup1,
+					true,
+					true,
+					false,
+					map[string]string{model.BackupTypeTag: model.AssistBackup},
+					nil),
+			},
+		},
+		{
+			name:  "Only assist bases",
+			input: testUser1Mail,
+			manifestData: []manifestInfo{
+				newManifestInfo(
+					testID2,
+					testT2,
+					testCompleteMan,
+					testBackup2,
+					nil,
+					testMail,
+					testUser1,
+				),
+				newManifestInfo(
+					testID1,
+					testT1,
+					testCompleteMan,
+					testBackup1,
+					nil,
+					testMail,
+					testUser1,
+				),
+			},
+			expectedBaseReasons: map[int][]Reasoner{},
+			expectedAssistManifestReasons: map[int][]Reasoner{
+				0: testUser1Mail,
+			},
+			expectedAssistReasons: map[int][]Reasoner{
+				0: testUser1Mail,
+			},
+			backupData: []backupInfo{
+				newBackupModel(
+					testBackup2,
+					true,
+					true,
+					false,
+					map[string]string{model.BackupTypeTag: model.AssistBackup},
+					nil),
+				newBackupModel(
+					testBackup1,
+					true,
+					true,
+					false,
+					map[string]string{model.BackupTypeTag: model.AssistBackup},
+					nil),
 			},
 		},
```

Comment:
I love the test coverage here since it tries a lot of different combinations. However, could you please add one more test case (mostly just a sanity check) to ensure we still get the expected merge base when the merge base is tagged with the new merge base tags (i.e. `map[string]string{model.BackupTypeTag: model.MergeBase}`)? The precise set of inputs/outputs don't matter so much so long as the base is selected

Reply:
Good suggestion! Done.